add-on/chrome/bootstrap.js

```diff
@@ -346,6 +348,132 @@ var WindowListener = {
         }
       },
 
+      // Allow the DNS service to be overwritten for testing.
+      _dns: Cc["@mozilla.org/network/dns-service;1"].getService(Ci.nsIDNSService),
+
+      /**
+       * Maybe add the copy panel if it's not throttled and passes other checks.
+       */
+      maybeAddCopyPanel() {
+        // Don't bother adding the copy panel if we're in private browsing or
+        // we've already shown it.
+        if (PrivateBrowsingUtils.isWindowPrivate(window) ||
+            Services.prefs.getBoolPref("loop.copy.shown")) {
+          return;
+        }
+
+        // Initialize the ticket (0-254) if it doesn't have a valid value yet.
+        let ticket = Services.prefs.getIntPref("loop.copy.ticket");
+        if (ticket < 0) {
+          ticket = Math.floor(Math.random() * COPY_MAX_TICKET);
+          Services.prefs.setIntPref("loop.copy.ticket", ticket);
+        }
+        // Short circuit if the special ticket value indicates we're good to go.
+        else if (ticket >= COPY_MAX_TICKET) {
+          this.addCopyPanel();
+          return;
+        }
+
+        // Look up the DNS A-record of a throttler hostname to decide to show.
+        this._dns.asyncResolve(
+          Services.prefs.getCharPref("loop.copy.throttler"),
+          this._dns.RESOLVE_DISABLE_IPV6,
+          (request, record) => {
```

Comment:
Moved to `let onDNS` with comments. Also collapsed the multiline-argument-style asyncResolve call to singleline-argument-style.